### PR TITLE
Legalize aten.repeat_interleave.Tensor for torch-mlir 

### PR DESCRIPTION
**Pull request by flemairen6**

### CODE REVIEW

File: python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/abstract_interp_lib_gen.py [hunk 1]
```diff
@@ -431,6 +431,9 @@ def aten〇repeat〡shape(self: List[int], repeats: List[int]) -> List[int]:
     for i in range(tensor_dim):
         out.append(self[i] * repeats[i + leading_rank])
     return out
+    
+def aten〇repeat_interleave〇Tensor〡shape(repeats: List[int], output_size: Optional[int] = None) -> List[int]:
+    return repeats
```
Comment (mgehre-amd) [inline python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/abstract_interp_lib_gen.py:436]: Isn't the output size somehow depending on the sum of entries in `repeats`? At least this is how I read https://pytorch.org/docs/stable/generated/torch.repeat_interleave.html#torch.repeat_interleave

File: python/torch_mlir_e2e_test/test_suite/basic.py [hunk 2]
```diff
@@ -1394,6 +1394,26 @@ def RepeatModule_basic(module, tu: TestUtils):
...
+    def __init__(self):
+        super().__init__()
+
+    @export
+    @annotate_args([
+        None,
+        ([3, 1, 2, 4], torch.int, True),
+    ])
+    def forward(self, x):
+        x = torch.ops.aten.repeat_interleave(x, output_size=10)
```
Comment (mgehre-amd) [inline python/torch_mlir_e2e_test/test_suite/basic.py:1408]:
```suggestion
        x = torch.ops.aten.repeat_interleave(x, output_size=10)
        y = torch.ops.aten.repeat_interleave(x)
        return x, y
```
Can you add another call without the optional output_size here?

**Review by mgehre-amd — CHANGES_REQUESTED** [review]

**Comment by mgehre-amd** [comment]
Seems like you need to mark the test as XFAIL for LTC

**Review by ttjost — APPROVED** [review]
Thanks!

**Comment by mgehre-amd** [comment]
Seems like you also have to mark linalg backend as xfail for now

**Review by mgehre-amd — APPROVED** [review]

**Pull request merged**